Source/WebCore/Modules/credentialmanagement/MediationRequirement.h

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if ENABLE(WEB_AUTHN)
+
+namespace WebCore {
+
+enum class MediationRequirement : uint8_t { Silent, Optional, Required, Conditional };
+
+} // namespace WebCore
+
+namespace WTF {
+
+template<> struct EnumTraits<WebCore::MediationRequirement> {
```

Comment:
I _think_ EnumTraits is no longer needed in favor of the new generated enum serialization mechanism, but I could be wrong

Reply:
As someone who contributes to WebKit ~once/year, can you tell me more about this? :)

Reply:
A new generator for serialization handles what the `EnumTraits` line here did. You can see an example of the change needed for `AttestationConveyancePreference` in https://github.com/WebKit/WebKit/pull/7694/files#diff-b7c9fc545913900a0089417beb52d397de465e8e1c044c1718efd2eb05307ab1. 

tl;dr: remove this line / everything in WTF namespace here. Then add 
`#if ENABLE(WEB_AUTHN)
enum class MediationRequirement : uint8_t { Silent, Optional, Required, Conditional };
#endif
`

in `WebCoreArgumentCoders.serialization.in`

Reply:
Neat, thank you!